docs/guides/validation.md

```diff
@@ -3,43 +3,113 @@ id: form-validation
 title: Form and Field Validation
 ---
 
-At the core of TanStack Form's functionalities is the concept of validation. We currently support three mechanisms of validation:
+At the core of TanStack Form's functionalities is the concept of validation. TanStack Form makes validation highly customizable:
+- You can control when to perform the validation (on change, on input, on blur, on submit...)
+- Validation rules can be defined at the field level or at the form level
+- Validation can be synchronous or asynchronous (for example as a result of an API call)
 
-- Synchronous functional validation
-- Asynchronous functional validation
-- Adapter-based validation
 
-Let's take a look at each and see how they're built.
+## When is validation performed?
 
-## Synchronous Functional Validation
+It's up to you! The `<Field />` component accepts some callbacks as props such as `onChange` or `onBlur`. Those callbacks are passed the current value of the field, as well as the fieldAPI object, so that you can perform the validation. If you find a validation error, simply return the error message as string and it will be available in `field.state.meta.errors`.
 
-With Form, you can pass a function to a field and, if it returns a string, said string will be used as the error:
+Here is an example:
+
+```tsx
+ <form.Field
+  name="firstName"
+  onChange={(value) => {
+    return !value
+      ? "A first name is required"
+      : value.length < 3
+        ? "First name must be at least 3 characters"
+        : undefined;
```

Comment:
This is admittedly our fault for having this example anywhere in our current docs; but people's names should not be validated for basically anything, especially length mins/maxes.

Let's change this validation example to something else - maybe back to our `age < 13` example? 😄

Reply:
I agree. I've updated all the examples to show an "age" field instead of "first name".